jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/EchoHandler.java

```diff
@@ -32,26 +31,32 @@ public class EchoHandler extends Handler.Abstract.NonBlocking
     public EchoHandler()
     {
     }
-    
+
     @Override
-    public boolean handle(Request request, Response response, Callback callback) throws Exception
+    public boolean handle(Request request, Response response, Callback callback)
     {
         response.setStatus(200);
-        String contentType = request.getHeaders().get(HttpHeader.CONTENT_TYPE);
-        if (StringUtil.isNotBlank(contentType))
-            response.getHeaders().put(HttpHeader.CONTENT_TYPE, contentType);
 
-        if (request.getHeaders().contains(HttpHeader.TRAILER))
+        long contentLength = -1;
+        for (HttpField field : request.getHeaders())
         {
-            HttpFields.Mutable responseTrailers = HttpFields.build();
-            response.setTrailersSupplier(() -> responseTrailers);
+            if (field.getHeader() != null)
+            {
+                switch (field.getHeader())
+                {
+                    case CONTENT_LENGTH ->
+                    {
+                        response.getHeaders().add(field);
+                        contentLength = field.getLongValue();
+                    }
+                    case CONTENT_TYPE -> response.getHeaders().add(field);
+                    case TRAILER -> response.setTrailersSupplier(HttpFields.build());
+                    case TRANSFER_ENCODING -> contentLength = Long.MAX_VALUE;
+                }
+            }
```

Comment:
Quite harder to read, because now different concerns are in the same code blob. 
-0 but I won't oppose too much.

Reply:
wow I find it so much easier to read as each concern is now only 1 or 2 lines and not spread over multiple blocks of code.